spring-web/src/main/java/org/springframework/http/HttpHeaders.java

```diff
@@ -781,6 +782,47 @@ public long getExpires() {
 	}
 
 	/**
+	 * Set the (new) value of the {@code Host} header.
+	 * <p>If the given {@linkplain InetSocketAddress#getPort() port} is {@code 0}, the host header
+	 * will only contain the {@linkplain InetSocketAddress#getHostString() hostname}.
+	 */
+	public void setHost(InetSocketAddress host) {
+		String value =
+				host.getPort() != 0 ? String.format("%s:%d", host.getHostString(), host.getPort()) :
+						host.getHostString();
+		set(HOST, value);
+	}
+
+	/**
+	 * Return the value of the required {@code Host} header.
+	 * <p>If the header value does not contain a port, the returned
+	 * {@linkplain InetSocketAddress#getPort() port} will be {@code 0}.
+	 */
+	public InetSocketAddress getHost() {
+		String value = getFirst(HOST);
+		if (value == null) {
+			return null;
+		}
+		int idx = value.indexOf(':');
```

Comment:
Will that work with IPv6 addresses though?
Wondering if `lastIndexOf` is enough or if more checks are required for addresses like `[2001:db8::ff00:42:8329]:8080`.